src/main/java/cz/xtf/wait/SupplierWaiter.java

```diff
@@ -0,0 +1,94 @@
+package cz.xtf.wait;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class SupplierWaiter<X> implements Waiter {
```

Comment:
Imho `WaiterSupplier` sounds a bit better or?

Reply:
`SupplierWaiter` - `Waiter` that uses `Supplier<X>` to retrieve state of object and those objects are tested for some state by `Function<X, boolean>`
`WaiterSupplier` - class that supplies waiters?

`SupplierWaiter` in my opinion better describes what to expected from class. Like, it's not an `Supplier<>` in java lambda manner.

Reply:
Ok, I do agree it might be more confusing with other `*Suppliers`. Consider this as resolved then :)